src/pages/SearchPage/SearchPage.js

```diff
@@ -26,39 +26,59 @@ function parseRedditData(input) {
     arrInfo[day][hour].push({
       author: result.author,
       title: result.title,
-      created_utc: result.created_utc,
+      created: date,
       score: result.score,
       num_comments: result.num_comments,
+      full_link: result.full_link,
     });
   });
   return arrInfo;
 }
 
 function SearchPage() {
-  const [isLoading, setIsLoading] = useState(true);
+  const [isLoading, setIsLoading] = useState(false);
   const [error, setError] = useState('');
   const [searchResults, setSearchResults] = useState([]);
   const [heatmapInfo, setHeatmapInfo] = useState([]);
-  const { subreddit } = useParams();
+  const { subreddit: currentPath } = useParams();
+  const history = useHistory();
 
-  // Search for new subreddit when parameters have changed
+  // Load data
+  //    Will load data if there:
+  //      1. isn't a current load in place,
+  //      2. is a change to history detected,
+  //      3. the useParams() property changes
   useEffect(() => {
-    setIsLoading(true);
-    setError('');
-    handleSearch(subreddit)
-      .then((data) => {
-        if (data.data.length !== 0) {
-          setSearchResults(data);
-        } else {
-          setError('0 results returned.');
-        }
-      })
-      .catch(() => {
-        setError('Error loading data.');
-      })
-      .finally(() => setIsLoading(false));
-  }, [subreddit]);
+    function loadData() {
+      const searchPath = history.location.pathname.match(/^\/\w*\/(\w*)$/i)[1] || 'javascript';
+      setIsLoading(true);
+      setError('');
 
+      handleSearch(searchPath)
+        .then((data) => {
+          if (data.data.length !== 0) {
+            setSearchResults(data);
+          } else {
+            setError('0 results returned.');
+          }
+        })
+        .catch(() => {
+          setError('Error loading data.');
+        })
+        .finally(() => setIsLoading(false));
+    }
+
+    if (!isLoading) {
+      loadData();
+    }
+
+    const unlisten = history.listen(() => {
+      loadData();
+    });
+    return unlisten;
+  }, [currentPath]);
```

Comment:
Btw I get this warning. Using `isLoading` inside useEffect and adding as a dependency is probably not a good idea since it will set the listener too often. I think it's actually OK to leave it out. You can deactivate the button when `isLoading` is true to prevent an update of the URL. And you can't prevent someone updating the URL manually. Wouldn't matter anyways since the page would be reloaded

![image](https://user-images.githubusercontent.com/9722792/78978709-9822b900-7b1a-11ea-85d2-8941973188bf.png)